Update Custom Redis versions documentation adding support for Relay (#2453)
platformsh/platformsh-docs: docs/src/languages/php/redis.md

[Redis](../../add-services/redis.md) is a popular structured key-value service, supported by Platform.sh.  It's frequently used for caching.

## Install PhpRedis

The [PhpRedis](https://github.com/phpredis/phpredis) extension is available on Platform.sh's PHP container images.  However, the extension has been known to break its API between versions when removing deprecated functionality.  The version available on each application image is the latest available at the time that PHP version was built, which if your application is very sensitive to PhpRedis versions may not be ideal.

> If the version of the PhpRedis extension available for your PHP version is not compatible with your application and upgrading your application is not feasible, you can use the script below as an alternative to download and compile a precise version of the extension on the fly.

Reviewer: :hammer_and_wrench: The script is no longer below, so it makes sense to me to change the wording.

```suggestion
If the version of the PhpRedis extension available for your PHP version is not compatible with your application
and upgrading your application is not feasible,
you can use the script linked below as an alternative to download and compile a precise version of the extension.
```

Do *not* use this approach unless you really need to.  Using the provided PhpRedis extension is preferred in the vast majority of cases.

> A [script](https://github.com/platformsh/snippets/blob/main/src/install-phpredis.sh) is provided to ease the installation of a customer version of PhpRedis.  Invoke that script from your build hook, specifying a version.  Any tagged version of the library is acceptable:

Reviewer: :hammer_and_wrench: We prefer links at the ends of sentences: https://github.com/platformsh/platformsh-docs/blob/main/contributing/content-style.md#link-at-the-end-of-sentences-in-sentence-case

```suggestion
To ease the installation of a customer version of PhpRedis, use a [PhpRedis install script](https://github.com/platformsh/snippets/blob/main/src/install-phpredis.sh).
Invoke this script from your build hook, specifying a version.
Any tagged version of the library is acceptable:
```

```yaml {location=".platform.app.yaml"}
hooks:
    build: |
        set -e
        # Install PhpRedis v5.1.1:
        curl -fsS https://raw.githubusercontent.com/platformsh/snippets/main/src/install-phpredis.sh | { bash /dev/fd/3 5.1.1 ; } 3<&0
```

## Install Relay

> Relay is a [Redis](../../add-services/redis.md) client like [PhpRedis](https://github.com/phpredis/phpredis) and [PhpRedis](https://github.com/predis/predis). It intents to be a drop-in replacement of those libraries.

Reviewer: :clipboard: Some small typos

```suggestion
Relay is a [Redis](../../add-services/redis.md) client
similar to [PhpRedis](https://github.com/phpredis/phpredis) and
[Predis](https://github.com/predis/predis).
It is intended to be a drop-in replacement for those libraries.
```

That PHP extension is also a shared in-memory cache like APCu. All retrieved keys are held in the PHP master process’ memory, which is shared across all FPM workers.

> That means if the FPM Worker #1 fetches the `users:count` key from Redis, the other FPM Workers #2, #3, … will instantaneously retrieve that key from Relay, without having to communicate with Redis.

Reviewer: :hammer_and_wrench: We prefer to avoid the future tense: https://github.com/platformsh/platformsh-docs/blob/main/contributing/content-style.md#use-the-present-tense

```suggestion
That means if the FPM Worker #1 fetches the `users:count` key from Redis,
all other FPM workers instantaneously retrieve that key from Relay without having to communicate with Redis.
```

> A [script](https://github.com/platformsh/snippets/blob/main/src/install-relay.sh) is provided to ease the installation of a customer version of PhpRedis.  Invoke that script from your build hook, specifying a version.  Any tagged version of the library is acceptable:

Reviewer: :hammer_and_wrench: Similar to above
```suggestion
To ease the installation of a customer version of Predis, use the [Predis install script](https://github.com/platformsh/snippets/blob/main/src/install-relay.sh).
Invoke this script from your build hook, specifying a version.
Any tagged version of the library is acceptable:
```

And either the example should be here or the colon changed to a period.

Reviewer 2: There is a typo in this suggestion:

```diff
-[Predis install script]
+[Relay install script]
```

Reviewer: Even more than that :facepalm: Better version:

```suggestion
To ease the installation of a customer version of Relay, use the [Relay install script](https://github.com/platformsh/snippets/blob/main/src/install-relay.sh).
Invoke this script from your build hook, specifying a version.
Any tagged version of the library is acceptable:
```

> ## Changing extension of version

Reviewer: :clipboard: I think the `of` was a typo and we prefer not to use gerunds in headings

```suggestion
## Change extension or version

```

> If you ever wish to change the Redis extension or the version you are using, update the build hook and clear the build cache: `platform project:clear-build-cache`.  The new version will *not* be used until you clear the build cache.

Reviewer: :hammer_and_wrench: We prefer to avoid the future tense: https://github.com/platformsh/platformsh-docs/blob/main/contributing/content-style.md#use-the-present-tense

```suggestion
To change the Redis extension or the version you are using, update the build hook and clear the build cache: `platform project:clear-build-cache`.
The new version is *not* be used until you clear the build cache.
```

There is no need to declare the extension in the `runtime` block.  That is only for pre-built extensions.

```yaml {location=".platform.app.yaml"}
hooks:
    build: |
        set -e
        # Install PhpRedis v0.4.3:
        curl -fsS https://raw.githubusercontent.com/platformsh/snippets/main/src/install-relay.sh | { bash /dev/fd/3 0.4.3 ; } 3<&0
```

> ## What those scripts do

Reviewer: :clipboard: 
```suggestion
## What these scripts do
```

> 1. Downloads the Relay/PhpRedis source code.
> 2. Checks out the version specified in the build hook.
> 3. Compiles the extension.
> 4. Copies the resulting `redis.so` file to [your app root](../../create-apps/app-reference.md#root-directory).

Reviewer: :clipboard: The scripts are plural

```suggestion
1. Download the Relay/PhpRedis source code.
2. Check out the version specified in the build hook.
3. Compile the extension.
4. Copy the resulting `redis.so` file to [your app root](../../create-apps/app-reference.md#root-directory).
```

Reviewer 2: Not sure if relevant, but Relay's shared object is called `relay.so`.

Reviewer: Certainly could make that clearer:

```suggestion
1. Download the Relay/PhpRedis source code.
2. Check out the version specified in the build hook.
3. Compile the extension.
4. Copy the resulting `relay.so`/`redis.so` file to [your app root](../../create-apps/app-reference.md#root-directory).
```
